lib/src/commands/admin.dart

```diff
@@ -1,112 +1,103 @@
+final admin = ChatGroup(
   'admin',
   'Administrative commands',
   children: [
     ChatCommand(
       'cleanup',
       'Bulk deletes messages in a channel',
       id('admin-cleanup', (
-        IChatContext context,
-        @UseConverter(IntConverter(min: 1))
-        @Description('The number of messages to delete')
-        int count, [
-        @Description('The user from whom to delete messages') IUser? user,
+        ChatContext context,
+        @UseConverter(IntConverter(min: 1)) @Description('The number of messages to delete') int count, [
+        @Description('The user from whom to delete messages') User? user,
       ]) async {
-        List<IMessage>? channelMessages;
+        List<Message>? channelMessages;
         Snowflake? last;
 
-        while (count > 0 &&
-            (channelMessages == null || channelMessages.isNotEmpty)) {
-          channelMessages = await context.channel
-              .downloadMessages(
-                limit: 100,
-                after: last,
-              )
-              .toList();
+        while (count > 0 && (channelMessages == null || channelMessages.isNotEmpty)) {
+          channelMessages = await context.channel.messages.fetchMany(
+            limit: 100,
+            after: last,
+          );
+
           last = channelMessages.last.id;
 
-          Iterable<IMessage> toRemove;
+          Iterable<Message> toRemove;
 
           if (user == null) {
             toRemove = channelMessages.take(count);
           } else {
-            toRemove = channelMessages
-                .where((message) => message.author.id == user.id);
+            toRemove = channelMessages.where((message) => message.author.id == user.id);
           }
 
-          try {
-            if (toRemove.length == 1) {
-              await toRemove.first.delete();
-            } else {
-              await context.channel.bulkRemoveMessages(toRemove);
-            }
-          } on IHttpResponseError catch (e) {
-            throw CheckedBotException(e.message);
+          if (toRemove.length == 1) {
+            await toRemove.first.delete();
+          } else {
+            await context.channel.messages.bulkDelete(toRemove.map((m) => m.id));
           }
 
           count -= toRemove.length;
         }
 
-        if (context is InteractionChatContext) {
-          await context.respond(
-              MessageBuilder.content('Successfully deleted messages!'),
-              hidden: true);
-        } else {
-          await context.respond(
-              MessageBuilder.content('Successfully deleted messages!'));
-        }
+        await context.respond(MessageBuilder(content: 'Successfully deleted messages!'));
       }),
-      checks: [PermissionsCheck(PermissionsConstants.manageMessages)],
+      checks: [PermissionsCheck(Permissions.manageMessages)],
       options: CommandsOptions(
-        hideOriginalResponse: true,
+        defaultResponseLevel: ResponseLevel.private,
       ),
```

Comment:
Should be moved to before the callback argument now we can put named args anywhere in a function invocation.

Reply:
how position of arguments affects how logic works? Also its weird to have named arguments before unnamed, required ones

Reply:
It doesn't affect the logic, but it's just nicer to put these parameters before the long callback parameter so they're clearly related to the invocation of the constructor, e.g: 
```dart
final command = ChatCommand(
  'name',
  'description',
  options: CommandOptions(...),
  (ChatContext context) {
    // A lot of code here.
  }
);
```